kv/kvserver: TestLargeUnsplittableRangeReplicate failed

kv/kvserver.TestLargeUnsplittableRangeReplicate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12087971?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12087971?buildTab=artifacts#/) on master @ [36601cc9b590061db2d821287985cbf53f4debd2](https://github.com/cockroachdb/cockroach/commits/36601cc9b590061db2d821287985cbf53f4debd2):


```
=== RUN   TestLargeUnsplittableRangeReplicate
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/be2c2653728c52ac0d08c8972ba7c01d/logTestLargeUnsplittableRangeReplicate2288097300
    test_log_scope.go:81: use -show-logs to present logs inline
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3,4,5}
    replicate_queue_test.go:1862: condition failed to evaluate within 45s: sql: no rows in result set
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/be2c2653728c52ac0d08c8972ba7c01d/logTestLargeUnsplittableRangeReplicate2288097300
--- FAIL: TestLargeUnsplittableRangeReplicate (60.58s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLargeUnsplittableRangeReplicate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32154

kv/kvserver.TestLargeUnsplittableRangeReplicate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12089374?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12089374?buildTab=artifacts#/) on master @ [d26a180f914dbb0a727566f1ac1f4f3ee4c624d9](https://github.com/cockroachdb/cockroach/commits/d26a180f914dbb0a727566f1ac1f4f3ee4c624d9):


```
=== RUN   TestLargeUnsplittableRangeReplicate
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/be2c2653728c52ac0d08c8972ba7c01d/logTestLargeUnsplittableRangeReplicate1118117561
    test_log_scope.go:81: use -show-logs to present logs inline
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3,4,5}
    replicate_queue_test.go:1862: condition failed to evaluate within 45s: sql: no rows in result set
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/be2c2653728c52ac0d08c8972ba7c01d/logTestLargeUnsplittableRangeReplicate1118117561
--- FAIL: TestLargeUnsplittableRangeReplicate (60.79s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLargeUnsplittableRangeReplicate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestLargeUnsplittableRangeReplicate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12147851?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12147851?buildTab=artifacts#/) on master @ [d399485b3d52934bda058b9e32f90b9ec09dcf22](https://github.com/cockroachdb/cockroach/commits/d399485b3d52934bda058b9e32f90b9ec09dcf22):


```
=== RUN   TestLargeUnsplittableRangeReplicate
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/be2c2653728c52ac0d08c8972ba7c01d/logTestLargeUnsplittableRangeReplicate3738913944
    test_log_scope.go:81: use -show-logs to present logs inline
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3,4,5}
    replicate_queue_test.go:1862: condition failed to evaluate within 45s: sql: no rows in result set
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/be2c2653728c52ac0d08c8972ba7c01d/logTestLargeUnsplittableRangeReplicate3738913944
--- FAIL: TestLargeUnsplittableRangeReplicate (60.82s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLargeUnsplittableRangeReplicate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestLargeUnsplittableRangeReplicate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12148336?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12148336?buildTab=artifacts#/) on master @ [d399485b3d52934bda058b9e32f90b9ec09dcf22](https://github.com/cockroachdb/cockroach/commits/d399485b3d52934bda058b9e32f90b9ec09dcf22):


```
=== RUN   TestLargeUnsplittableRangeReplicate
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/be2c2653728c52ac0d08c8972ba7c01d/logTestLargeUnsplittableRangeReplicate3071927
    test_log_scope.go:81: use -show-logs to present logs inline
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3,4,5}
    replicate_queue_test.go:1862: condition failed to evaluate within 45s: sql: no rows in result set
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/be2c2653728c52ac0d08c8972ba7c01d/logTestLargeUnsplittableRangeReplicate3071927
--- FAIL: TestLargeUnsplittableRangeReplicate (60.93s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112207 kv/kvserver: TestLargeUnsplittableRangeReplicate failed [C-test-failure O-robot T-kv branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLargeUnsplittableRangeReplicate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestLargeUnsplittableRangeReplicate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12156538?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12156538?buildTab=artifacts#/) on master @ [b071093ec762b195eec20889621544e416dc0edf](https://github.com/cockroachdb/cockroach/commits/b071093ec762b195eec20889621544e416dc0edf):


```
=== RUN   TestLargeUnsplittableRangeReplicate
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/be2c2653728c52ac0d08c8972ba7c01d/logTestLargeUnsplittableRangeReplicate1754504301
    test_log_scope.go:81: use -show-logs to present logs inline
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3}
    replicate_queue_test.go:1854: replicas: {1,2,3,4,5}
    replicate_queue_test.go:1862: condition failed to evaluate within 45s: sql: no rows in result set
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/be2c2653728c52ac0d08c8972ba7c01d/logTestLargeUnsplittableRangeReplicate1754504301
--- FAIL: TestLargeUnsplittableRangeReplicate (60.76s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112207 kv/kvserver: TestLargeUnsplittableRangeReplicate failed [C-test-failure O-robot T-kv branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLargeUnsplittableRangeReplicate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -1811,6 +1811,21 @@ func TestLargeUnsplittableRangeReplicate(t *testing.T) {
 	toggleReplicationQueues(tc, true /* active */)
 	toggleSplitQueues(tc, true /* active */)
 
+	// Check that the two ranges exist for table t.
+	testutils.SucceedsSoon(t, func() error {
+		r := db.QueryRow(
+			"SELECT count(*) FROM [SHOW RANGES FROM TABLE t]")
+		var count int
+		if err := r.Scan(&count); err != nil {
+			return err
+		}
+		if count != 2 {
+			return fmt.Errorf(
+				"splits not created, expected %d ranges, found %d", 2, count)
+		}
+		return nil
+	})
+
 	// We're going to create a large row, but now large enough that write
 	// back-pressuring kicks in and refuses it.
 	var sb strings.Builder

